nixos/modules/services/misc/autobrr.nix

```diff
@@ -0,0 +1,76 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.autobrr;
+  configFormat = pkgs.formats.toml { };
+  configTemplate = configFormat.generate "autobrr.toml" cfg.settings;
+in {
+  options = {
+    services.autobrr = {
+      enable = mkEnableOption "Autobrr";
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description =
+          "Open ports in the firewall for the Autobrr web interface.";
+      };
+
+      secretFile = mkOption {
+        type = types.path;
+        description =
+          "File containing the session secret for the Autobrr web interface.";
+      };
+
+      settings = lib.mkOption {
+        type = lib.types.submodule { freeformType = configFormat.type; };
+        default = {
+          host = "127.0.0.1";
+          port = "7474";
+          checkForUpdates = true;
+        };
+        example = { logLevel = "DEBUG"; };
+        description = ''
+          Autobrr configuration options.
+
+          Refer to <https://autobrr.com/configuration/autobrr>
+          for a full list.
+        '';
+      };
+
+      package = mkPackageOption pkgs "autobrr" { };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [{
+      assertion = cfg.settings.sessionSecret == null;
+      message = "Session secrets must be passed using the secretFile option.";
+    }];
+
+    systemd.services.autobrr = {
+      description = "Autobrr";
+      after = [ "syslog.target" "network-online.target" ];
+      wants = [ "network-online.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = {
+        Type = "simple";
+        DynamicUser = true;
+        StateDirectory = "autobrr";
+        StateDirectoryMode = "0700";
+        ExecPreStart =
+          "${lib.getExe pkgs.dasel} put -f '${configTemplate}' -v '${
+            builtins.readFile cfg.secretFile
+          }' -o %S/config.toml 'sessionSecret'";
```

Comment:
FYI ``serviceConfig.EnvironmentFile = cfg.secretFile;`` might work better

Reply:
Also if nix is reading the file doesn't it go into nix store? (Didn't test somebody correct me if not)

Reply:
You might be right about builtins.readFile adding files to the Nix store (I am also not sure), but I have used shell substitution instead. I don't want to use the environment variables because they must be retyped by users in the `EnvironmentFile`, and systemd doesn't have a way to prevent users from specifying settings in the environment which conflict with those in the module's configuration.

Reply:
I also cannot find any documentation for autobrr's environment variables outside of the code, so I don't want to point users towards them.

Reply:
https://autobrr.com/installation/docker#environment-variables
They are mentioned here in the documentation. Also from what i have seen, env files for secrets are a common pattern in nixpkgs

Reply:
Anyway, apart from my opinion on this, looks good now 👍